ui/src/shared/components/LayoutCellHeader.tsx

```diff
@@ -1,34 +1,86 @@
-import React, {SFC} from 'react'
+import React, {Component} from 'react'
 import classnames from 'classnames'
+import chroma from 'chroma-js'
 import {isCellUntitled} from 'src/dashboards/utils/cellGetters'
+import {DEFAULT_CELL_BG_COLOR} from 'src/dashboards/constants'
+import {ErrorHandling} from 'src/shared/decorators/errors'
 
 interface Props {
   isEditable: boolean
   cellName: string
   makeSpaceForCellNote: boolean
+  cellBackgroundColor: string
+  cellTextColor: string
 }
 
-const LayoutCellHeader: SFC<Props> = ({
-  isEditable,
-  cellName,
-  makeSpaceForCellNote,
-}) => {
-  const headingClass = `dash-graph--heading ${
-    isEditable ? 'dash-graph--draggable dash-graph--heading-draggable' : ''
-  }`
-
-  return (
-    <div className={headingClass}>
-      <span
-        className={classnames('dash-graph--name', {
-          'dash-graph--name__default': isCellUntitled(cellName),
-          'dash-graph--name__note': makeSpaceForCellNote,
-        })}
-      >
+@ErrorHandling
+class LayoutCellHeader extends Component<Props> {
+  public render() {
+    return (
+      <div className={this.headingClass}>
+        {this.cellName}
+        {this.headingBar}
+      </div>
+    )
+  }
+
+  private get headingClass(): string {
+    const {isEditable} = this.props
+
+    return classnames('dash-graph--heading', {
+      'dash-graph--draggable': isEditable,
+      'dash-graph--heading-draggable': isEditable,
+    })
+  }
+
+  private get cellName(): JSX.Element {
+    const {
+      cellName,
+      makeSpaceForCellNote,
+      cellTextColor,
+      cellBackgroundColor,
+    } = this.props
+
+    const className = classnames('dash-graph--name', {
+      'dash-graph--name__default': isCellUntitled(cellName),
+      'dash-graph--name__note': makeSpaceForCellNote,
+    })
+
+    let nameStyle = {}
+
+    if (cellBackgroundColor !== DEFAULT_CELL_BG_COLOR) {
+      nameStyle = {
+        color: cellTextColor,
+      }
+    }
+
+    return (
+      <span className={className} style={nameStyle}>
         {cellName}
       </span>
-    </div>
-  )
+    )
+  }
+
+  private get headingBar(): JSX.Element {
+    const {isEditable, cellBackgroundColor} = this.props
+
+    let barStyle = {}
+
+    if (cellBackgroundColor !== DEFAULT_CELL_BG_COLOR) {
```

Comment:
I think it would make more immediate sense to reader if this was in the isEditable if block- but this works as well so feel free to keep it.